my_sprites.py

```diff
@@ -1,4 +1,87 @@
 import arcade
+import math
+
+class Enemy(arcade.Sprite):
+    """
+    parent class for all enemies in the game
+    """
+
+    def __init__(self, filename: str, center_pos: tuple[float, float], max_hp: int, speed: float, scale=1.0):
+
+        super().__init__(
+            filename=filename,
+            scale=scale,
+            center_x=center_pos[0],
+            center_y=center_pos[1]
+        )
+
+        # hp
+        self._max_hp = max_hp
+        self.cur_hp = max_hp
+
+        self.speed = speed
+
+        # pathfinding
+        self.path = []
+        self.cur_path_position = 0  # which point on the path we are heading for.
+
+    @property
+    def max_hp(self):
+        return self._max_hp
+
+    def find_path(self, barrier_list: arcade.AStarBarrierList, target_pos: tuple[int, int]):
+        """
+        calculates a path to the target pos. Sets the sprite's path to this path.
+        If an enemy has a path, it will automatically follow it.
+        """
+
+        # calculate the path. It will be a list of positions(lists)
+        self.path = arcade.astar_calculate_path(self.position,
+                                                target_pos,
+                                                barrier_list,
+                                                diagonal_movement=True)
+
+        # reset this because we are at the start of a new path
+        self.cur_path_position = 0
+
+    def on_update(self, delta_time: float = 1 / 60):
+
+        # follow the path, if present
+        if self.path:
+
+            # next position to move to
+            dest_pos = self.path[self.cur_path_position]
+
+            # calculate angle to next point
+            angle_to_dest = arcade.get_angle_radians(dest_pos[0], dest_pos[1], self.center_x, self.center_y)
+
+            # calculate distance
+            distance_to_dest = arcade.get_distance(dest_pos[0], dest_pos[1], self.center_x, self.center_y)
+
+            # this is so we don't move too far
+            this_move = min(self.speed, distance_to_dest)
+
+            # if we are there, set the next position to move to
+            if distance_to_dest <= self.speed:
```

Comment:
I think it's missing the last step towards the destination this way. Since you newer overshoot the destination, using var _this_move_, I think you would end up with a distance if 0 (if you round off).

You can keep this, but if you do, you really don't need var _this_move_, and you accept that the enemy will never reach the exact destination.